server.go

```diff
@@ -312,6 +314,9 @@ type Server struct {
 	DecorateWriter DecorateWriter
 	// Maximum number of TCP queries before we close the socket. Default is maxTCPQueries (unlimited if -1).
 	MaxTCPQueries int
+	// Whether to set the SO_REUSEPORT socket option, allowing multiple listeners to be bound to a single address.
+	// It is only supported on go1.11+ and when using ListenAndServe.
+	Reuseport bool
```

Comment:
@miekg Are you happy with naming this `Reuseport bool`?

Reply:
sgtm, maybe ReusePort (to nit about it)

otherwise this is all lgtm, feel free to merge.